R/layer_threshold_preds.R

```diff
@@ -1,8 +1,16 @@
 #' Lower and upper thresholds for predicted values
 #'
-#' This postprocessing step is used to set prediction values that are
-#' smaller than the lower threshold or higher than the upper threshold equal
-#' to the threshold values.
+#' This post-processing step is used to set prediction values that are smaller
+#' than the lower threshold or higher than the upper threshold equal to the
+#' threshold values.
+
+#' @details
+#' Making case count predictions strictly positive is a typical example usage.
+#'   It can be called before or after the quantiles are created using
+#'   `layer_quantile_distn()` since the quantiles are an inherent part of the
+#'   result from `layer_predict()` for distributional models, but must be called
+#'   after `layer_residual_quantiles()`, since the quantiles for that case don't
+#'   exist until after that layer.
```

Comment:
This is confusing to me. It's location in the frosting must be such that the
variable in `...` has been created by an earlier layer.

Reply:
It may not be worth documenting it, but since `layer_predict()` creates a quantile in `.pred`, the threshold just needs to be after `layer_predict()` for a quantile distribution; for example, this works just fine:
```r
jhu <- covid_case_death_rates %>%
  filter(time_value > "2021-11-01", geo_value %in% c("ak", "ca", "ny"))

r <- epi_recipe(jhu) %>%
  step_epi_lag(death_rate, lag = c(0, 7, 14)) %>%
  step_epi_ahead(death_rate, ahead = 7) %>%
  step_epi_naomit()

wf <- epi_workflow(r, quantile_reg(quantile_levels = c(.25, .5, .75))) %>%
  fit(jhu)

f <- frosting() %>%
  layer_predict() %>%
  layer_threshold(.pred, lower = 1.180, upper = 3.310) %>%
  layer_quantile_distn() %>%
  layer_naomit(.pred)
wf1 <- wf %>% add_frosting(f)

p <- forecast(wf1)
p
```

Reply:
probably more robust to default to recommending adding it late and using `starts_with(".pred")`